lib/publiccloud/gcp.pm

```diff
@@ -36,6 +36,7 @@ sub vault_gcp_roles {
 sub init {
     my ($self) = @_;
     $self->SUPER::init();
+    $self->vault(publiccloud::vault->new());
```

Comment:
PR is saying about AWS refactoring . is this part necessary to do now ?

Reply:
Yes, the creation of the vault instance was defined on provider.pm see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/55d00d82a6aeba43fd0ceb3b9dcd2ecb078106ba/lib/publiccloud/provider.pm#L45 but this will be moved to each client (aws_client and gcp_client) for now google is in the old way, so It should be maintained